src/presenter/details.js

```diff
@@ -0,0 +1,88 @@
+define(["../model/stops", "../model/departures", "../view/details"],
+function (stops, departures, view) {
+    console.log("Details module loaded");
+
+    // It allows us to navigate through the history without request it again
+    var lastHash = "";
+
+    // In case of the user back to the search view before the result has come
+    var canceledFetch = false;
+
+    // Callback executed when the AJAX finishes
+    function successStops() {
+        if (canceledFetch) {
+            return;
+        }
+        var values = stops.getStops();
+        view.renderStops(values);
+        view.appendStops();
+    }
+
+    // Callback executed when the AJAX finishes
+    function successDepartures() {
+        if (canceledFetch) {
+            return;
+        }
+        var values = departures.getDepartures();
+        view.renderDepartures(values);
+        view.appendDepartures();
+    }
+
+    // Loads details (stops and departures) fetching them by id
+    function load(id) {
+        console.log("[details] Load id: " + id);
+        canceledFetch = false;
+        stops.fetch(id, successStops);
+        departures.fetch(id, successDepartures);
+    }
+
+    // When back button is pressed
+    view.on("click", function () {
+        canceledFetch = true;
+        window.history.back();
+    });
+
+    function navigate() {
```

Comment:
I have saved only the last search. If you navigate through forward and backward button you can see that the request is not triggered again. I preserve the state in the HTML elements. I didn't save some request because I thought that was not a requirement. My fault!